spec/src/main/asciidoc/microprofile-lra-spec.adoc

```diff
@@ -731,6 +732,110 @@ Only when the implementation is certain that participant has finished
+long running action and uses a Java 8 completion stage to return
+an asynchronous response:
+
+[source,java]
+----
+    @LRA(value = LRA.Type.REQUIRED, // the method must run with an LRA
+         end = true) // the LRA must end when the method completes
+    @Path("completion-stage-lra")
+    public CompletionStage<Response> asyncInvocationWithLRA(
+        @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) { // the id of LRA
+
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        final CompletableFuture<Response> response = new CompletableFuture<>();
+
+        try {
+            es.submit(() -> {
+                // excecute a long running business activity and resume when done
+                ...
+                response.complete(Response.ok().entity(lraId).build());
+                // since the response is complete the implemention will
+                // close the LRA thereby invoking any registered participant
+                // completion handlers
+            });
+        } finally {
+            es.shutdown();
+        }
+
+        return response;
+    }
+----
+
+With completion stages it is also possible to complete exceptionally. The
+following example should run business logic asynchronously in the context
+of an LRA but the LRA should be cancelled: forcing any registered
+participant compensation handlers to run:
+
+[source,java]
+----
+    @LRA(value = LRA.Type.REQUIRED, // the method must run with an LRA
+         end = true, // the LRA must end when the method completes
+         cancelOn = {Response.Status.NOT_FOUND}) // cancel on a 404 code
+    @Path("completion-stage-exceptionally-lra")
+    public CompletionStage<Response> asyncInvocationWithException(
+        @HeaderParam(LRA_HTTP_CONTEXT_HEADER) String lraId) {
+
+        ExecutorService es = Executors.newSingleThreadExecutor();
+        final CompletableFuture<Response> response = new CompletableFuture<>();
+
+        try {
+            es.submit(() -> {
+                // excecute long running business activity finishing with a NOT_FOUND error
+                // is one of the default LRA#cancelOn codes that should cause the LRA to cancel
+                response.completeExceptionally(
+                        new WebApplicationException(
+                            Response.status(Response.Status.NOT_FOUND).entity(lraId).build()));
+            });
+        } finally {
+            es.shutdown();
+        }
+
+        return response;
+    }
+----
+
+In addition to the use of completion stages, a resource method may also produce
+asynchronous responses by injecting a JAX-RS `AsyncResponse` parameter using
+the JAX-RS `@Suspended` annotation:
+
+[source,java]
+----
+    @LRA(value = LRA.Type.REQUIRED, // the method must run with an LRA
+         end = true) // the LRA must end when the method completes
+    @Path("asyncresponse-lra")
+    public CompletionStage<Response> asyncResponseWithLRA(
```

Comment:
seems this is forgotten previous method definition

Reply:
+1